src/compiler/checker.ts

```diff
@@ -40419,6 +40466,9 @@ namespace ts {
         }
 
         function isAliasResolvedToValue(symbol: Symbol): boolean {
+            if (symbol === unknownSymbol) {

```

Comment:
Why does a `source` of `unknownSymbol` return `false`, but a `target` of `unknownSymbol` returns `true`?

Also, `resolveAlias` has an assertion that you should have already checked whether `symbol` has `SymbolFlags.Alias`, so that check should also have been done prior to calling `isAliasResolvedToValue`, in which case you shouldn't get `unknownSymbol` since it never has `SymbolFlags.Alias`.

Reply:
If you look up at `isValueAliasDeclaration`, it clearly knowingly calls `isAliasResolvedToValue` with `unknownSymbol`, but it seems that code path was never hit or tested, since it would cause an assertion error. It seemed to me like the expectation was that `isAliasResolvedToValue` should be tolerant of `unknownSymbol`, but I can change that assumption instead.

Reply:
The question is whether a `source` of `unknownSymbol` should not be considered a value alias (i.e., return `false`) when a `target` of `unknownSymbol` *is* considered a value alias (i.e., returns `true`). I think the idea was, if we don't know then we assume its a value when it comes to elision, etc, in which case `source === unknownSymbol` probably should also return `true`.